go-controller/pkg/libovsdbops/dhcp.go

```diff
@@ -0,0 +1,71 @@
+package libovsdbops
+
+import (
+	"reflect"
+
+	libovsdbclient "github.com/ovn-org/libovsdb/client"
+	libovsdb "github.com/ovn-org/libovsdb/ovsdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+)
+
+type dhcpOptionsPredicate func(*nbdb.DHCPOptions) bool
+
+func CreateOrUpdateDhcpOptionsOps(nbClient libovsdbclient.Client, ops []libovsdb.Operation, lsp *nbdb.LogicalSwitchPort, dhcpv4Options, dhcpv6Options *nbdb.DHCPOptions) ([]libovsdb.Operation, error) {
+	opModels := []operationModel{}
+	if dhcpv4Options != nil {
+		opModels = append(opModels, operationModel{
+			Model: dhcpv4Options,
+			ModelPredicate: func(item *nbdb.DHCPOptions) bool {
+				return item.Cidr == dhcpv4Options.Cidr && reflect.DeepEqual(item.ExternalIDs, dhcpv4Options.ExternalIDs)
+			},
+			OnModelUpdates: onModelUpdatesAllNonDefault(),
+			DoAfter:        func() { lsp.Dhcpv4Options = &dhcpv4Options.UUID },
+			ErrNotFound:    false,
+			BulkOp:         false,
+		})
+	}
+	if dhcpv6Options != nil {
+		opModels = append(opModels, operationModel{
+			Model: dhcpv6Options,
+			ModelPredicate: func(item *nbdb.DHCPOptions) bool {
+				return item.Cidr == dhcpv6Options.Cidr && reflect.DeepEqual(item.ExternalIDs, dhcpv6Options.ExternalIDs)
+			},
+			OnModelUpdates: onModelUpdatesAllNonDefault(),
+			DoAfter:        func() { lsp.Dhcpv6Options = &dhcpv6Options.UUID },
+			ErrNotFound:    false,
+			BulkOp:         false,
+		})
+	}
+	opModels = append(opModels, operationModel{
+		Model:          lsp,
+		ModelPredicate: func(item *nbdb.LogicalSwitchPort) bool { return item.Name == lsp.Name },
+		OnModelUpdates: onModelUpdatesAllNonDefault(),
```

Comment:
for extra carefulness I would only update the Dhcpv4Options or Dhcpv6Options columns